src/Common/OpenTelemetryTraceContext.cpp

```diff
@@ -231,63 +247,75 @@ TracingContextHolder::TracingContextHolder(
     const Settings * settings_ptr,
     const std::weak_ptr<OpenTelemetrySpanLog> & _span_log)
 {
-    if (current_thread_trace_context.isTraceEnabled())
+    /// Use try-catch to make sure the ctor is exception safe.
+    /// If any exception is raised during the construction, the tracing is not enabled on current thread.
+    try
```

Comment:
(cosmetic and just FYI for future commits: C++ has a little known feature called [function try blocks](https://en.cppreference.com/w/cpp/language/function-try-block) which would have been a good fit here)

Reply:
Thanks for the information.